### PR TITLE
Merge20231204

### PR DESCRIPTION
**Pull request by samtertzakian**
Merge20231204

1. Add support to DMF_IoctHandler to allow multiple instances of the same IOCTL table to be attached to same WDFDEVICE. It allows two instances of a Parent module that has DMF_IoctlHanlder as a Child Module to be instantiated in the same driver such that ReferenceSring is used to route WDFREQUEST to the proper instance.
2. Fix Verifier issue in DMF_NotifyUserWithRequestMultiple relaated to untagged dereference.
3. Fix some SAL issues.
4. Add DMF_ConnectedStandby Module.

### CODE REVIEW

File: Dmf/Modules.Library/Dmf_ConnectedStandby.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+## DMF_ConnectedStandby
+
+-----------------------------------------------------------------------------------------------------------------------------------
+
+#### Module Summary
+
+-----------------------------------------------------------------------------------------------------------------------------------
+
+This Module allows the Client to request and receive notifications when the system has entered or exited Connected Standby.
```
Comment (ankurung) [inline Dmf/Modules.Library/Dmf_ConnectedStandby.md:9]: Windows 10 Modern Standby (Modern Standby) expands the Windows 8.1 Connected Standby power model

File: Dmf/Modules.Library/Dmf_ConnectedStandby.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+--*/
+{
+    NTSTATUS ntStatus;
+    DMF_CONTEXT_ConnectedStandby* moduleContext;
+
+    PAGED_CODE();
+
+    FuncEntry(DMF_TRACE);
+
+    ntStatus = STATUS_SUCCESS;
```
Comment (ankurung) [inline Dmf/Modules.Library/Dmf_ConnectedStandby.c:259]: unnecessary.

File: Dmf/Modules.Library/Dmf_ConnectedStandby.c [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// WDF Module Callbacks
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+
+#pragma code_seg("PAGE")
+NTSTATUS
+ConnectedStandby_OnConnectedStandby(
+    LPCGUID SettingGuid,
```
[the remark this answers is not in the document]
Comment (samtertzakian) [reply Dmf/Modules.Library/Dmf_ConnectedStandby.c]: fixed

File: Dmf/Modules.Library/Dmf_ConnectedStandby.md (hunk 1 above)
Comment (samtertzakian) [reply Dmf/Modules.Library/Dmf_ConnectedStandby.md]: fixed

File: Dmf/Modules.Library/Dmf_ConnectedStandby.c [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+{
+    NTSTATUS ntStatus;
+    DMF_CONTEXT_ConnectedStandby* moduleContext;
+
+    PAGED_CODE();
+
+    FuncEntry(DMF_TRACE);
+
+    moduleContext = DMF_CONTEXT_GET(DmfModule);
+
```
[the remark this answers is not in the document]
Comment (samtertzakian) [reply Dmf/Modules.Library/Dmf_ConnectedStandby.c:209]: fixed

File: Dmf/Modules.Library/Dmf_ConnectedStandby.c (hunk 2 above)
Comment (samtertzakian) [reply Dmf/Modules.Library/Dmf_ConnectedStandby.c]: fixed

**Review by ankurung — APPROVED** [review]

**Pull request merged**